### ISSUE TITLE
PK-CLI: Polykey Agent Start Timeout Error in latest release 

### ISSUE THREAD

**Issue by CryptoTotalWar** [MatrixAI/Polykey-CLI#218]
### To Reproduce

1. I got the error where `polykey agent start -v` starts a loop of messages and ends with failed to send data on the QUICSocket, ending the loop. 
2. I then downloaded the PK-CLI repo and ran it from there, getting the same error 
```
QUICClientInternal: QUIC Client internal error - Failed to send data on the QUICSocket
ErrorQUICClientInternal: Failed to send data on the QUICSocket
```

### Expected behavior

running Polykey agent start in the foreground should not start what feels like an infinity loop that times out, it should be a stable stream of messages that feels stable. 

### Screenshots

This is just 1 small portion of the entire terminal output from the `polykey agent start` command. 

![image](https://github.com/MatrixAI/Polykey-CLI/assets/107008759/1b446c93-9f5f-4c01-977c-4aba0ec0e66e)

### Platform (please complete the following information)
 - Device: MAC OS 
 - Version:
 
 I ran this with the PK version i have on my machine and with the staging branch of the PK-CLI repo. 

### Additional context

https://discord.com/channels/1137566877506359436/1137567759518142574/1255617291970936985

### Notify maintainers

@tegefaulkes

**Comment by tegefaulkes** [comment]
I'm assuming this is the latest version of the CLI then? Please be explicit about that, include the version used.

**Comment by CryptoTotalWar** [comment]
This is the node version on my machine. 

```
version                 1.5.1-1-1
sourceVersion           1.5.1
stateVersion            1
networkVersion          1
versionMetadata  
  version               0.4.2
  commitHash            "1554e81cc28739be22fc2cf92c7f2b5c01643a63\n"
  libVersion            1.5.1-1-1
  libSourceVersion      1.5.1
  libStateVersion       1
  libNetworkVersion     1
rulerpablo@pablos-air Polykey-CL
```

From the build i do off the staging branch on the repo, I can't even view the status. The only command I can run from the build off the staging branch is` node ./dist/polykey.js agent start --background` or `node ./dist/polykey.js agent start -v` in the latter case i get the following output:

```
 rulerpablo@pablos-air Polykey-CLI % node ./dist/polykey.js agent start -v                                                18:22:34
✔ Please enter the password … ******
INFO:polykey.PolykeyAgent:Creating PolykeyAgent
INFO:polykey.PolykeyAgent:Setting umask to 077
INFO:polykey.PolykeyAgent:Setting node path to /Users/rulerpablo/Library/Application Support/polykey
INFO:polykey.PolykeyAgent.Status:Starting Status
INFO:polykey.PolykeyAgent.Status:Writing Status to /Users/rulerpablo/Library/Application Support/polykey/status.json
INFO:polykey.PolykeyAgent.Status:Status is STARTING
INFO:polykey.PolykeyAgent.Schema:Creating Schema
INFO:polykey.PolykeyAgent.Schema:Starting Schema
INFO:polykey.PolykeyAgent.Schema:Setting state path to /Users/rulerpablo/Library/Application Support/polykey/state
INFO:polykey.PolykeyAgent.Schema:Started Schema
INFO:polykey.PolykeyAgent.Schema:Created Schema
INFO:polykey.PolykeyAgent.KeyRing:Creating KeyRing
INFO:polykey.PolykeyAgent.KeyRing:Setting keys path to /Users/rulerpablo/Library/Application Support/polykey/state/keys
INFO:polykey.PolykeyAgent.KeyRing:Starting KeyRing
INFO:polykey.PolykeyAgent.KeyRing:Checking /Users/rulerpablo/Library/Application Support/polykey/state/keys/private.jwk
INFO:polykey.PolykeyAgent.KeyRing:Loading root key pair
INFO:polykey.PolykeyAgent.KeyRing:Checking /Users/rulerpablo/Library/Application Support/polykey/state/keys/db.jwk
INFO:polykey.PolykeyAgent.KeyRing:Started KeyRing
INFO:polykey.PolykeyAgent.KeyRing:Created KeyRing
INFO:polykey.PolykeyAgent.DB:Creating DB
INFO:polykey.PolykeyAgent.DB:Starting DB
INFO:polykey.PolykeyAgent.DB:Setting DB path to /Users/rulerpablo/Library/Application Support/polykey/state/db
INFO:polykey.PolykeyAgent.DB:Started DB
INFO:polykey.PolykeyAgent.DB:Created DB
INFO:polykey.PolykeyAgent:Creating TaskManager
INFO:polykey.PolykeyAgent:Starting TaskManager in Lazy Mode
INFO:polykey.PolykeyAgent:Begin Tasks Repair
WARN:polykey.PolykeyAgent:Moving Task v0pjsiahijto017m5a0g3u4d4m4 from Active to Queued
WARN:polykey.PolykeyAgent:Moving Task v0pjsib175do010igr0d4uumos4 from Active to Queued
WARN:polykey.PolykeyAgent:Moving Task v0pjsib17glo0126g3lssm2m26c from Active to Queued
WARN:polykey.PolykeyAgent:Moving Task v0pjsiehnglo012iqr4nbb3b02c from Active to Queued
WARN:polykey.PolykeyAgent:Moving Task v0pjsiehnq9o011jm5ho8vk2nb0 from Active to Queued
WARN:polykey.PolykeyAgent:Moving Task v0pjsif28kdo010ao7nbc4lmoqo from Active to Queued
WARN:polykey.PolykeyAgent:Moving Task v0pjsif28u5o01dver3id8drbpc from Active to Queued
WARN:polykey.PolykeyAgent:Moving Task v0pjsih0631o01793l1af5kim28 from Active to Queued
WARN:polykey.PolykeyAgent:Moving Task v0pjsih06cpo012f8eirfufch70 from Active to Queued
WARN:polykey.PolykeyAgent:Moving Task v0pjsiikrhlo012ceumdvsnoqpo from Active to Queued
INFO:polykey.PolykeyAgent:Finish Tasks Repair
INFO:polykey.PolykeyAgent:Started TaskManager
INFO:polykey.PolykeyAgent:Created TaskManager
INFO:polykey.PolykeyAgent.CertManager:Creating CertManager
INFO:polykey.PolykeyAgent.CertManager:Starting CertManager
INFO:polykey.PolykeyAgent.CertManager:Begin current certificate setup
INFO:polykey.PolykeyAgent.CertManager:Existing current certificate found
INFO:polykey.PolykeyAgent.CertManager:Finish current certificate setup
INFO:polykey.PolykeyAgent.CertManager:Registering handler CertManager.renewCurrentCert
INFO:polykey.PolykeyAgent.CertManager:Started CertManager
INFO:polykey.PolykeyAgent.CertManager:Created CertManager
INFO:polykey.PolykeyAgent.Sigchain:Creating Sigchain
INFO:polykey.PolykeyAgent.Sigchain:Starting Sigchain
INFO:polykey.PolykeyAgent.Sigchain:Started Sigchain
INFO:polykey.PolykeyAgent.Sigchain:Created Sigchain
INFO:polykey.PolykeyAgent.ACL:Creating ACL
INFO:polykey.PolykeyAgent.ACL:Starting ACL
INFO:polykey.PolykeyAgent.ACL:Started ACL
INFO:polykey.PolykeyAgent.ACL:Created ACL
INFO:polykey.PolykeyAgent.GestaltGraph:Creating GestaltGraph
INFO:polykey.PolykeyAgent.GestaltGraph:Starting GestaltGraph
INFO:polykey.PolykeyAgent.GestaltGraph:Started GestaltGraph
INFO:polykey.PolykeyAgent.GestaltGraph:Created GestaltGraph
INFO:polykey.PolykeyAgent.IdentitiesManager:Creating IdentitiesManager
INFO:polykey.PolykeyAgent.IdentitiesManager:Starting IdentitiesManager
INFO:polykey.PolykeyAgent.IdentitiesManager:Started IdentitiesManager
INFO:polykey.PolykeyAgent.IdentitiesManager:Created IdentitiesManager
INFO:polykey.PolykeyAgent.NodeGraph:Creating NodeGraph
INFO:polykey.PolykeyAgent.NodeGraph:Starting NodeGraph
INFO:polykey.PolykeyAgent.NodeGraph:Started NodeGraph
INFO:polykey.PolykeyAgent.NodeGraph:Created NodeGraph
INFO:polykey.PolykeyAgent.Discovery:Creating Discovery
INFO:polykey.PolykeyAgent.Discovery:Starting Discovery
INFO:polykey.PolykeyAgent.Discovery:Started Discovery
INFO:polykey.PolykeyAgent.Discovery:Created Discovery
INFO:polykey.PolykeyAgent.NotificationsManager:Creating NotificationsManager
INFO:polykey.PolykeyAgent.NotificationsManager:Starting NotificationsManager
INFO:polykey.PolykeyAgent.NotificationsManager:Started NotificationsManager
INFO:polykey.PolykeyAgent.NotificationsManager:Created NotificationsManager
INFO:polykey.PolykeyAgent.VaultManager:Creating VaultManager
INFO:polykey.PolykeyAgent.VaultManager:Setting vaults path to /Users/rulerpablo/Library/Application Support/polykey/state/vaults
INFO:polykey.PolykeyAgent.VaultManager:Starting VaultManager
INFO:polykey.PolykeyAgent.VaultManager.EFS Database:Creating DB
INFO:polykey.PolykeyAgent.VaultManager.EFS Database:Starting DB
INFO:polykey.PolykeyAgent.VaultManager.EFS Database:Setting DB path to /Users/rulerpablo/Library/Application Support/polykey/state/vaults/efs
INFO:polykey.PolykeyAgent.VaultManager.EFS Database:Started DB
INFO:polykey.PolykeyAgent.VaultManager.EFS Database:Created DB
INFO:polykey.PolykeyAgent.VaultManager.EncryptedFileSystem.INodeManager:Creating INodeManager
INFO:polykey.PolykeyAgent.VaultManager.EncryptedFileSystem.INodeManager:Starting INodeManager
INFO:polykey.PolykeyAgent.VaultManager.EncryptedFileSystem.INodeManager:Started INodeManager
INFO:polykey.PolykeyAgent.VaultManager.EncryptedFileSystem.INodeManager:Created INodeManager
INFO:polykey.PolykeyAgent.VaultManager.EncryptedFileSystem:Starting EncryptedFS
INFO:polykey.PolykeyAgent.VaultManager.EncryptedFileSystem:Started EncryptedFS
INFO:polykey.PolykeyAgent.VaultManager:Started VaultManager
INFO:polykey.PolykeyAgent.VaultManager:Created VaultManager
INFO:polykey.PolykeyAgent.SessionManager:Creating SessionManager
INFO:polykey.PolykeyAgent.SessionManager:Starting SessionManager
INFO:polykey.PolykeyAgent.SessionManager:Started SessionManager
INFO:polykey.PolykeyAgent.SessionManager:Created SessionManager
INFO:polykey.PolykeyAgent.Audit:Creating Audit
INFO:polykey.PolykeyAgent.Audit:Starting Audit
INFO:polykey.PolykeyAgent.Audit:Started Audit
INFO:polykey.PolykeyAgent.Audit:Created Audit
INFO:polykey.PolykeyAgent:Starting PolykeyAgent
INFO:polykey.PolykeyAgent.ClientService:Start ClientService
INFO:polykey.PolykeyAgent.ClientService.RPCServer:Start RPCServer
INFO:polykey.PolykeyAgent.ClientService.RPCServer:Started RPCServer
INFO:polykey.PolykeyAgent.ClientService.WebSocketServer:Starting WebSocketServer
INFO:polykey.PolykeyAgent.ClientService.WebSocketServer:Started WebSocketServer
INFO:polykey.PolykeyAgent.ClientService:Started ClientService
INFO:polykey.PolykeyAgent.NodeConnectionManager:Start NodeConnectionManager on [::]:0
INFO:polykey.PolykeyAgent.NodeConnectionManager.RPCServer:Start RPCServer
INFO:polykey.PolykeyAgent.NodeConnectionManager.RPCServer:Started RPCServer
INFO:polykey.PolykeyAgent.NodeConnectionManager.QUICSocket:Start QUICSocket on [::]:0
INFO:polykey.PolykeyAgent.NodeConnectionManager.QUICSocket:Started QUICSocket on [::]:54532
INFO:polykey.PolykeyAgent.NodeConnectionManager.QUICServer:Start QUICServer on [::]:54532
INFO:polykey.PolykeyAgent.NodeConnectionManager.QUICServer:Started QUICServer on [::]:54532
INFO:polykey.PolykeyAgent.NodeConnectionManager:Started NodeConnectionManager
INFO:polykey.PolykeyAgent.NodeManager:Starting NodeManager
INFO:polykey.PolykeyAgent.NodeManager:Setting up refreshBucket tasks
INFO:polykey.PolykeyAgent.NodeManager:Set up refreshBucket tasks
INFO:polykey.PolykeyAgent.NodeManager.MDNS:Start MDNS
INFO:polykey.PolykeyAgent.NodeManager.MDNS:Started MDNS
INFO:polykey.PolykeyAgent.NodeManager:Started NodeManager
WARN:polykey.PolykeyAgent.NodeManager:Duplicate refreshBucket task was found for bucket 255, cancelling
WARN:polykey.PolykeyAgent.NodeManager:Duplicate refreshBucket task was found for bucket 253, cancelling
INFO:polykey.PolykeyAgent.NodeManager.syncNodeGraph:Synchronizing NodeGraph
INFO:polykey.PolykeyAgent.scheduler:Starting Scheduling Loop
INFO:polykey.PolykeyAgent.scheduler:Started Scheduling Loop
INFO:polykey.PolykeyAgent.queue:Starting Queueing Loop
INFO:polykey.PolykeyAgent.queue:Started Queueing Loop
INFO:polykey.PolykeyAgent.WorkerManager:Creating WorkerManager
INFO:polykey.PolykeyAgent.WorkerManager:Created WorkerManager
INFO:polykey.PolykeyAgent.Status:Finish Status STARTING
INFO:polykey.PolykeyAgent.Status:Writing Status to /Users/rulerpablo/Library/Application Support/polykey/status.json
INFO:polykey.PolykeyAgent.Status:Status is LIVE
INFO:polykey.PolykeyAgent:Started PolykeyAgent
INFO:polykey.PolykeyAgent:Created PolykeyAgent
pid             40835
nodeId          vgijtpv0h8m1eajeir77g73muq88n5kj0413t6fjdqsv9kt8dq4pg
clientHost      ::1
clientPort      59256
agentHost       ::
agentPort       54532
INFO:polykey.PolykeyAgent.NodeConnectionManager.NodeConnectionForward [3.145.86.40:1314]:Creating forward NodeConnection
INFO:polykey.PolykeyAgent.NodeConnectionManager.NodeConnectionForward [3.145.86.40:1314].QUICClient:Create QUICClient to 3.145.86.40:1314
INFO:polykey.PolykeyAgent.NodeConnectionManager.NodeConnectionForward [2600:1f16:1f71:7c00:4aa5:9c9d:57f5:a649:1314]:Creating forward NodeConnection
INFO:polykey.PolykeyAgent.NodeConnectionManager.NodeConnectionForward [2600:1f16:1f71:7c00:4aa5:9c9d:57f5:a649:1314].QUICClient:Create QUICClient to [2600:1f16:1f71:7c00:4aa5:9c9d:57f5:a649]:1314
INFO:polykey.PolykeyAgent.NodeConnectionManager.NodeConnectionForward [3.145.86.40:1314].QUICClient.QUICConnection 85c0abd30ba0a4b5d859b4b9edc863fecc526d11:Connect QUICConnection
INFO:polykey.PolykeyAgent.NodeConnectionManager.NodeConnectionForward [2600:1f16:1f71:7c00:4aa5:9c9d:57f5:a649:1314].QUICClient.QUICConnection 59515bc16a84a2d8563fc4a7efb41db7a85654ca:Connect QUICConnection
INFO:polykey.PolykeyAgent.NodeConnectionManager.NodeConnectionForward [3.145.86.40:1314].QUICClient.QUICConnection 85c0abd30ba0a4b5d859b4b9edc863fecc526d11:Start QUICConnection
INFO:polykey.PolykeyAgent.NodeConnectionManager.NodeConnectionForward [2600:1f16:1f71:7c00:4aa5:9c9d:57f5:a649:1314].QUICClient.QUICConnection 59515bc16a84a2d8563fc4a7efb41db7a85654ca:Start QUICConnection
INFO:polykey.PolykeyAgent.NodeConnectionManager.NodeConnectionForward [2600:1f16:1f71:7c00:4aa5:9c9d:57f5:a649:1314].QUICClient:ErrorQUICClientInternal: QUIC Client internal error - Failed to send data on the QUICSocket
ErrorQUICClientInternal: Failed to send data on the QUICSocket
rulerpablo@pablos-air Polykey-CLI %                          
```

**Comment by tegefaulkes** [comment]
You're running on an old version of the CLI. You have `0.4.2` when the newest is `0.5.5`. Since that version change @amydevs has fixed a problem with `IPv6` connection but also the seed nodes are running on `IPv6` as well now. So the problem is likely the old version here.

Use the newer version and try again.

**Comment by tegefaulkes** [comment]
I think part of the problem here is that failure to send on a socket is a connection error. It should be handled internally and ignored. The fact that it bubbles up to the top of the process is a problem.

**Comment by amydevs** [comment]
this should be fixed in 0.5.5 @tegefaulkes @CryptoTotalWar, the place where this bubbles is on the `NodeConnectionManager.holePunch` method

**Comment by CryptoTotalWar** [comment]
Thanks, I'm not seeing the error anymore and everything seems to be working fine for me. I also had another user install & bootstrap PK on their machine and they didn't seem to be getting any of those same issues either. 

I'll be closing this ticket now as the bug issue has been resolved. 

However, @tegefaulkes  

Roger mentioned `Create issue for ensuring stability between staging and master and released builds by running tests on the built binaries prior to releasing.` 

and that this issue should be in  `1.0.0`
